src/Kestrel/WebHostBuilderKestrelExtensions.cs

```diff
@@ -23,10 +26,11 @@ public static class WebHostBuilderKestrelExtensions
         /// </returns>
         public static IWebHostBuilder UseKestrel(this IWebHostBuilder hostBuilder)
         {
-            hostBuilder.UseLibuv();
-
             return hostBuilder.ConfigureServices(services =>
             {
+                // Don't override an already-configured transport
+                services.TryAddSingleton<ITransportFactory, SocketTransportFactory>();
```

Comment:
Why not call UseSockets()

Reply:
Calling UseSockets here would cause UseKestrel override any previously configured transport. This is different than the old behavior with the libuv default, but it makes the TransportSelectors easier to implement.